packages/x-internals/src/cancellableEvent/index.ts

```diff
@@ -3,3 +3,7 @@ export type MuiCancellableEvent = {
 };
 
 export type MuiCancellableEventHandler<Event> = (event: Event & MuiCancellableEvent) => void;
+
+export const shouldSkipEventHandler = (event: MuiCancellableEvent): boolean => {
```

Comment:
I'd be interested to have your opinion on the name of this method.

My logic behind having it instead of accessing `.defaultMuiPrevented` directly is that we will probably have a `.defaultBasePrevented` at some point and having a util that check both will be nice.

Reply:
I'm unsure about this assumption. 🤔 
I think that on `MUI X` packages we could have `defaultMuiPrevented` and on `Base UI X` it could be `defaultBasePrevented`, but when consuming this library internally, we could add the `defaultMuiPrevented` prop in those cases.
This would allow us to keep the boolean event flag without creating a need for this function. 🤔 
I'm always vary of extra functions increasing the call stack. 🙈 😆 

WDYT? 🤔

Reply:
But the MUI X version will be built on top of the Base UI X version.
And all the Base UI X code will check for `.defaultMuiPrevented`, not for `.defaultMuiPrevented`.

So someone using the MUI X version will likely not have a correct behavior if he uses `.defaultMuiPrevented`, unless we re-implement all the prevention in the MUI X wrapper (not scalable at all IMHO) or if the Base UI X also checks for `.defaultMuiPrevented` (super ugly :grimacing: )

I guess the core team will have the same topic, not sure if they discussed it...
One could say that we should just all migrate to `.defaultBasePrevented` or even rename it to something more generic.

Reply:
If we implement a solution that means any given package always have to check only on attribute on the event, then I agree that the method is not needed.
I'll remove it from this PR since it's value is hypothetical for no.

Reply:
Maybe we could ask @mui/material-ui if they have thought about this? 🤔 

> I guess the core team will have the same topic, not sure if they discussed it...
One could say that we should just all migrate to .defaultBasePrevented or even rename it to something more generic.

This is a great idea. 👍 
It would be nice to know the Base and Material plan in this regard. 🤞 
cc @mui/base-ui I see that there are [usages of `defaultMuiPrevented`](https://github.com/search?q=repo%3Amui%2Fbase-ui%20defaultMuiPrevented&type=code) in your codebase.
Is there any plan to change it? Or is it here to stay?

Reply:
This is a leftover from @mui/base. We now use a different pattern that relies on a closure and does not require adding a field to the event object.

Reply:
That way, we can start discussing how it should look like on MUI X / Base UI X :pray:

Reply:
But do you know how Material UI will prevent the user logic?

Reply:
@michaldudak Oh, ok, this is new. The history of `.defaultMuiPrevented`

1. 2019: https://github.com/mui/material-ui/pull/17941/files#diff-804dd912f2ccdef39076fb6213816137debd3d39c0f32e940792bdaec8efbb8cR335. One event that bubble up between two different swipeable drawer. The first one that catches the event excludes the others with: `event.muiHandled = true;` (as the event bubble, the value is retained).
2. 2020: https://github.com/mui/material-ui/issues/19887. We had a different problem to solve, about allowing developers to opt-out of native event logic. We copied downshift https://github.com/mui/material-ui/issues/19887#issuecomment-619559387 which also happened to be the same as 1.
3. 2021: The data grid was using stopPropagation() for the same use case as 2. it was a mess, we copied the solution from 2. https://github.com/mui/mui-x/issues/1403
4. 2024: `event.preventBaseUIHandler();` 

I think we should align 2. 3. and 4. to be the same. As for 1. it's a private API, so doesn't matter.

Reply:
If we name the method `preventXGridHandler` (or `preventXHandler` but one per project can be nice for very specific use case like when you have a picker inside a grid), then we don't have any naming issue between MUI X and Base UI X :ok_hand:

Reply:
With a 2nd thought on it, I fear that it will be hard for X to migrate to a method like `preventXGridHandler` because of the lesser unified structure of our codebase.

I'd be in favor of not introducing any new complexity and just rename the property `.defaultXPrevented` or `.defaultXGridPrevented` if we want a per product property.

If we go for `.defaultPrevented` we keep the interfaces in the shared package but remove my method which will be useless.
If we go for `.defaultXGridPrevented` then we don't put anything in the shared package

Reply:
Updated

- no more logic in `@mui/x-internals`
- exported variable are renamed to prepare for the renaming of the property

Follow up:

- rename `defaultMuiPrevented` into `defaultXTreeViewPrevented` (can be done during alpha)